routing/index_graph_starter.cpp

```diff
@@ -4,190 +4,98 @@
 
 namespace routing
 {
-IndexGraphStarter::IndexGraphStarter(IndexGraph & graph, RoadPoint const & startPoint,
-                                     RoadPoint const & finishPoint)
-  : m_graph(graph)
-  , m_start(graph, startPoint, graph.GetNumJoints())
-  , m_finish(graph, finishPoint, graph.GetNumJoints() + 1)
+IndexGraphStarter::IndexGraphStarter(IndexGraph & graph, FakeVertex const & start,
+                                     FakeVertex const & finish)
+  : m_graph(graph), m_start(start, false /* backward */), m_finish(finish, true /* forward */)
 {
-  m_start.SetupJointId(graph);
-
-  if (startPoint == finishPoint)
-    m_finish.m_jointId = m_start.m_jointId;
-  else
-    m_finish.SetupJointId(graph);
 }
 
-m2::PointD const & IndexGraphStarter::GetPoint(Joint::Id jointId)
+m2::PointD const & IndexGraphStarter::GetPoint(Segment const & segment, bool isOutgoing)
 {
-  if (jointId == m_start.m_fakeId)
-    return m_graph.GetGeometry().GetPoint(m_start.m_point);
+  if (segment == m_start.GetFakeSegment())
+    return m_start.GetPoint();
 
-  if (jointId == m_finish.m_fakeId)
-    return m_graph.GetGeometry().GetPoint(m_finish.m_point);
+  if (segment == m_finish.GetFakeSegment())
+    return m_finish.GetPoint();
 
-  return m_graph.GetPoint(jointId);
+  return m_graph.GetGeometry().GetPoint(segment.GetRoadPoint(isOutgoing));
 }
 
-m2::PointD const & IndexGraphStarter::GetPoint(RoadPoint const & rp)
+// static
+size_t IndexGraphStarter::GetRouteNumPoints(vector<Segment> const & route)
 {
-  return m_graph.GetPoint(rp);
+  // if route contains start and finish fakes only, it doesn't have segment points.
+  // TODO: add start and finish when RecounstructRoute will be reworked.
+  if (route.size() <= 2)
```

Comment:
What if start point is the same as the finish point? It's a corner, but a legal case, and the method should return 1, isn't it?